array_family/flex_ext.py

```diff
@@ -751,6 +751,24 @@ def iterate_experiments_and_indices(self, experiments):
         for experiment, indices in zip(experiments, index_list):
             yield experiment, indices
 
+    def random_split(self, n=2):
+        """Randomly split table into n tables.
+
+        Not all tables will be the same length."""
+        n = int(n)
+        if n <= 1:
+            return [self]
+        cctbx.array_family.flex.set_random_seed(0)
```

Comment:
I'm not sure this should be here. It may be preferable to set once at a higher level, because setting here may cause unintended side effects from calling this function. If there is other code that uses the flex RNG later in the same process, its behaviour may depend on whether `random_split` was called or not.